File: VB/DXWebApplication1/Controllers/HomeController.vb
Imports System.Web
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc
'...


Namespace DXWebApplication1.Controllers
    Public Class HomeController
        Inherits AsyncController

        Public Function Index() As ActionResult
            ViewBag.Message = "Welcome to DevExpress Extensions for ASP.NET MVC!"
            Return View()
        End Function


        Public Function DocumentViewerPartial() As ActionResult
            Return PartialView("DocumentViewerPartial")
        End Function


        Public Function ExportDocumentViewer() As ActionResult
            ' Specify the report service URI and the name of the remote report to export.     
            Dim settings As New MVCxDocumentViewerRemoteSourceSettings()
            settings.ServerUri = "http://localhost:52511/ReportService1.svc"
            settings.ReportTypeName = "WebApplication1.XtraReport1, WebApplication1"

            Return DocumentViewerExtension.ExportTo(settings, HttpContext.Request)
        End Function
    End Class
End Namespace

File: CS/DXWebApplication1/Controllers/HomeController.cs
using System.Web;
using System.Web.Mvc;
using DevExpress.Web.Mvc;
//...


namespace DXWebApplication1.Controllers
{
    public class HomeController : AsyncController
    {
        public ActionResult Index()
        {
            ViewBag.Message = "Welcome to DevExpress Extensions for ASP.NET MVC!";
            return View();
        }


        public ActionResult DocumentViewerPartial() {
            return PartialView("DocumentViewerPartial");
        }
        

        public ActionResult ExportDocumentViewer() {
            // Specify the report service URI and the name of the remote report to export.            
            MVCxDocumentViewerRemoteSourceWcfReportServiceSettings settings = 
                new MVCxDocumentViewerRemoteSourceWcfReportServiceSettings();
            settings.ServiceUri = @"http://localhost:53813//ReportService1.svc";
            settings.ReportTypeName = "WebApplication1.XtraReport1, WebApplication1";

            return DocumentViewerExtension.ExportTo(settings, HttpContext.Request);
        }
 
    
    }
}